core/app/models/spree/taxon.rb

```diff
@@ -79,12 +79,32 @@ def to_param
       permalink
     end
 
+    # @return [Array<Integer>] the ids of descendant taxons.
+    def descendant_ids
+      descendants.pluck(:id)
+    end
+
     # @return [ActiveRecord::Relation<Spree::Product>] the active products the
     #   belong to this taxon
     def active_products
       products.not_deleted.available
     end
 
+    # @return [ActiveRecord::Relation<Spree::Product>] all self and descendant products
+    def all_products
+      Product.joins(:taxons).where(spree_taxons: { id: descendant_ids.push(id) })
+    end
+
+    # @return [ActiveRecord::Relation<Spree::Variant>] all self and descendant variants, including master variants.
+    def all_variants
```

Comment:
@cbrunsdon I will change that, my original thinking for the other way was to save database calls but it only saves us one db call. Thank you for your feedback.